core/src/main/java/org/radargun/SlaveBase.java

```diff
@@ -73,6 +74,17 @@ protected void scenarioLoop() throws IOException {
             } else if (!stage.shouldExecute()) {
                log.info("Stage '" + stage.getName() + "' should not be executed");
                response = new DistStageAck(state);
+               if (stage instanceof AbstractDistStage) {
```

Comment:
I guess I could call Cluster.groupExists() after the AbstractDistStage.shouldExecute() call. I'll look at that tomorrow!